apps/email/js/html_cache_restore.js

```diff
@@ -443,12 +443,29 @@ window.appDispatchedMessage = false;
 
     if (pendingUiMessageType) {
       startupData.entry = pendingUiMessageType;
-    } else if (navigator.mozHasPendingMessage('request-sync')) {
-      // While request-sync is not important for the pendingUiMessageType
+    } else if (navigator.mozHasPendingMessage('alarm')) {
+      // While alarm is not important for the pendingUiMessageType
       // gateway, it still should be indicated that the entry point was not the
       // default entry point, so that the UI is not fully started if this is a
       // background sync.
-      startupData.entry = 'request-sync';
+      startupData.entry = 'alarm';
+    } else if (navigator.mozHasPendingMessage('request-sync')) {
+      // Legacy request-sync trigger. Just clear all request-sync tasks, and
+      // allow the UI to go to a normal view this time. When back end starts up
+      // it will ensure the sync alarms are set properly for the next sync, so
+      // in this legacy case we will just miss one sync pass.
+      var navSync = navigator.sync;
+      if (navSync) {
+        navSync.registrations().then(function(regs) {
+          regs.forEach(function(reg) {
+console.log('*****DEBUG: unregistering request sync!');
+            reg.unregister(reg.task);
```

Comment:
ugh i think you are right, updating now. looking back at my log, i only saw the unregistering message. i will do that retest too.